db/migrate/20210414203512_convert_primary_language_to_iso_code.rb

```diff
@@ -0,0 +1,145 @@
+          note = "Primary language was listed as '#{key}' \
+                which could not be matched to the standard language list. This monitoree’s primary \
+                language has been updated to '#{value.nil? ? 'blank' : Languages.all_languages[value.to_sym][:display]}'. You \
+                may update this value."
+          execute <<-SQL.squish
+                INSERT INTO histories (patient_id,created_by,comment,history_type,created_at,updated_at)
+                SELECT id, 'Sara Alert System', "#{note}", 'System Note', CURRENT_TIMESTAMP, CURRENT_TIMESTAMP
+                FROM patients
+                WHERE purged = false AND primary_language = "#{key}"
+          SQL
+          Patient.where(purged: false).where(primary_language: key).update_all(primary_language: value)
+
+          note = "Secondary language was listed as '#{key}' \
+                which could not be matched to the standard language list. This monitoree’s secondary \
+                language has been updated to '#{value.nil? ? 'blank' : Languages.all_languages[value.to_sym][:display]}'. You \
+                may update this value."
+          execute <<-SQL.squish
+                INSERT INTO histories (patient_id,created_by,comment,history_type,created_at,updated_at)
+                SELECT id, 'Sara Alert System', "#{note}", 'System Note', CURRENT_TIMESTAMP, CURRENT_TIMESTAMP
+                FROM patients
+                WHERE purged = false AND secondary_language = "#{key}"
+          SQL
+          Patient.where(purged: false).where(secondary_language: key).update_all(secondary_language: value)
+        end
+      end
+    end
+  end
+
+  def down
+    execute <<-SQL.squish
+        UPDATE patients SET primary_language = legacy_primary_language, secondary_language = legacy_secondary_language
+    SQL
+
+    remove_column :patients, :legacy_primary_language
+    remove_column :patients, :legacy_secondary_language
+
+    # Destroy all System Notes that contain the text `language was listed as`
+    History.where('history_type = ? AND comment like ?', 'System Note', '%language was listed as%').destroy_all
+  end
+
+  def match_language(lang)
+    return nil if lang.nil?
+
+    lang = lang.to_s.downcase.strip
+    matched_language = nil
+    matched_language = lang.to_sym if Languages.all_languages[lang.to_sym].present?
+    return matched_language unless matched_language.nil?
+
+    matched_language = Languages.all_languages.find { |_key, val| val[:display]&.casecmp(lang)&.zero? }
+    # [:fra, {:display=>"French", :iso6391code=>"fr", :system=>"iso639-2t" }]
+    # matched_language will take the form of the above, and we want to return the 3-letter code at [0]
+    return matched_language[0] unless matched_language.nil?
+
+    matched_language = Languages.all_languages.find { |_key, val| val[:iso6391code]&.casecmp(lang)&.zero? }
+    return matched_language[0] unless matched_language.nil?
+
+    matched_language
```

Comment:
this looks like it's going to return `nil` if it can't match anything, did we want to default to `eng` for `primary_language` if we aren't able to successfully find any matches or convert it to `nil`?

Reply:
Nope, we want to make it `nil`. If it is `nil`, then it will choose english when it selects the language

Reply:
I want to bring this back up. @DPC15038 can you elaborate on what you mean by "when it selects the language"?

What I am observing when running the migration is that Patients with an unknown language are getting empty string as their language rather than English.

Reply:
That is the goal. We want to set it to empty.


If we ever send Patients assessments, and their primary_language is empty, it'll choose English there. We wouldnt want to change 'GIBBERISH' to 'English' cause they dont speak English. So we just set it to null

Reply:
Ok that answers my question, thanks